cmd/telliot/cmds.go

```diff
@@ -470,84 +535,90 @@ func (m mineCmd) Run() error {
 
 	// Not using a remote DB so need to start the trackers.
 	if cfg.Mine.RemoteDBHost == "" {
-		ds, err = ops.CreateDataServerOps(ctx, logger, cfg, proxy, client, contract, account, ch1)
+		ds, err = dataServer.CreateDataServerOps(ctx, logger, cfg, proxy, client, contract, accounts)
 		if err != nil {
 			return errors.Wrapf(err, "creating data server")
 		}
 		// Start and wait for it to be ready.
-		if err := ds.Start(ctx); err != nil {
-			return errors.Wrapf(err, "starting data server")
+		if err := ds.Start(); err != nil {
+			return errors.Wrap(err, "starting data server")
 		}
+		// We need to wait until the DataServer instance is ready.
 		<-ds.Ready()
 	}
 
-	var v []byte
-	for i := 0; i < 10; i++ {
-		// Start miner
-		v, err = proxy.Get(db.DisputeStatusKey)
-		if err != nil {
-			level.Warn(logger).Log("msg", "getting dispute status. Check if staked", "err", err)
-		}
-		if len(v) != 0 {
-			break
-		}
-		select {
-		case <-c: // Early exit from os.Interrupt
-			return nil
-		default:
+	// We define our run groups here.
+	var g run.Group
+	// Run groups.
+	{
+		// Handle interupts.
+		g.Add(run.SignalHandler(context.Background(), syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM))
+
+		// Metrics server.
+		{
+			http.Handle("/metrics", promhttp.Handler())
+			srv := &http.Server{Addr: fmt.Sprintf("%s:%d", cfg.Mine.ListenHost, cfg.Mine.ListenPort)}
+			g.Add(func() error {
+				level.Info(logger).Log("msg", "starting metrics server", "addr", cfg.Mine.ListenHost, "port", cfg.Mine.ListenPort)
+				// returns ErrServerClosed on graceful close
+				var err error
+				if err = srv.ListenAndServe(); err != http.ErrServerClosed {
+					err = errors.Wrapf(err, "ListenAndServe")
+				}
+				return err
+			}, func(error) {
+				srv.Close()
+			})
 		}
-		time.Sleep(1 * time.Second)
-	}
-	if len(v) == 0 {
-		return errors.New("no status result after 10 attempts. this usually means no connection to the DB")
-	}
 
-	status, _ := hexutil.DecodeBig(string(v))
-	if status.Cmp(big.NewInt(1)) != 0 {
-		return errors.New("miner is not able to mine with current status")
-	}
-	ch2 := make(chan os.Signal)
-	exitChannels = append(exitChannels, &ch2)
-	miner, err := ops.CreateMiningManager(logger, ch2, cfg, proxy, contract, account)
-	if err != nil {
-		return errors.Wrapf(err, "creating miner")
-	}
-	go func() {
-		miner.Start(ctx)
-	}()
-
-	// Wait for kill sig.
-	<-c
-	// Then notify exit channels.
-	for _, ch := range exitChannels {
-		*ch <- os.Interrupt
-	}
-	cnt := 0
-	start := time.Now()
-	for {
-		cnt++
-		dsStopped := false
-		minerStopped := false
-
-		if ds != nil {
-			dsStopped = !ds.Running
-		} else {
-			dsStopped = true
-		}
+		// Run a miner manager for each of the accounts.
+		if true {
+			// Run a tasker instance.
+			tasker, taskerCh := tasker.CreateTasker(ctx, logger, cfg, proxy, clientWs, contract, accounts)
```

Comment:
I think we also need a takser instance for each account as well. 
Otherwise the takser sends the tasks for each account on the same channel which means that each miner will mine the solutions for all accounts and then also each submitter will submit the solutions for all accounts. Ping me if I didn't manage to explain the problem properly.

Reply:
I manage to do it using one tasker using a map of tasker channels.